zenoh/src/api/session.rs

```diff
@@ -1574,19 +1574,31 @@ impl SessionInner {
                                     },
                                 }),
                             });
+                            #[cfg(feature = "unstable")]
+                            {
+                                let state = zread!(self.state);
+                                self.update_matching_status(
+                                    &state,
+                                    &sub_state.key_expr,
+                                    MatchingStatusType::Subscribers,
+                                    false,
+                                )
+                            }
+                        } else {
+                            drop(state);
```

Comment:
This `drop` is not necessary per se, as the code is at the end of the function, but I think it's a good thing to keep it, as it would avoid further error

Reply:
Removed it, I found it ugly 😅